# Day 23:30 daily statistics - MRLM - P1

$\color{red}{\text{Exemplo 8.2}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** Consideremos o seguinte conjunto de dados:

|    Y    |   X  |
|:-------:|:----:|
|  7,9682 | 0,86 |
|  2,3753 | 0,45 |
|  4,1545 | 0,85 |
|  3,5966 | 0,59 |
|  6,7885 | 0,88 |
|  2,1096 | 0,51 |
| 27,9359 | 1,82 |
| -0,9106 | 0,16 |
| -0,8716 | 1,38 |
| 16,7971 | 1,38 |
|  6,4503 | 0,69 |
|  5,2800 | 0,65 |
|  2,9231 | 0,45 |
|  0,4823 | 0,33 |
|  1,7878 | 0,38 |

+ a) Ajuste um modelo de regressão linear simples;
+ b) Incorpore um termo ao quadrado ao modelo;

$\color{red}{\text{a)}}$ - **Ajuste um modelo de regressão linear simples**

In [1]:
#Entrando com os dados
exemplo=read.csv2("exemplo8.2.csv",header=T,dec=",")

#Ajustando o modelo escolhido - Linear Simples
ajuste1=lm(Y~X,data=exemplo)
summary(ajuste1)


Call:
lm(formula = Y ~ X, data = exemplo)

Residuals:
     Min       1Q   Median       3Q      Max 
-14.6422  -0.2582   0.8597   1.0413   8.5146 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   -3.952      2.516  -1.570 0.140318    
X             12.843      2.868   4.478 0.000622 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 4.896 on 13 degrees of freedom
Multiple R-squared:  0.6067,	Adjusted R-squared:  0.5764 
F-statistic: 20.05 on 1 and 13 DF,  p-value: 0.000622


O modelo estimado é dado por $\hat{y}=-3,952+12,843X$. Pelo p-valor pode-se concluir que o modelo ajustado é altamente significativo. 

$\color{red}{\text{b)}}$ - **Incorpore um termo ao quadrado ao modelo**

In [2]:
#criando a variável X ao quadrado
X2=exemplo$X^2

#Adicionando uma nova coluna
exemplo=cbind(exemplo,X2)

#Ajustando um modelo
ajuste2=lm(Y~X+X2,data=exemplo)
summary(ajuste2)


Call:
lm(formula = Y ~ X + X2, data = exemplo)

Residuals:
     Min       1Q   Median       3Q      Max 
-14.0009  -0.5738   0.7726   2.4368   4.4304 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)
(Intercept)    2.103      4.700   0.447    0.663
X             -3.832     11.454  -0.335    0.744
X2             8.567      5.714   1.499    0.160

Residual standard error: 4.677 on 12 degrees of freedom
Multiple R-squared:  0.6687,	Adjusted R-squared:  0.6135 
F-statistic: 12.11 on 2 and 12 DF,  p-value: 0.001321


O novo modelo estimado é dado por $\hat{y}=2,103-3,832X+8,567X^2$. O modelo possui um coeficiente de determinação de $R^2=66,87\%$.

$\color{red}{\text{Exercício 1}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** Para estudar a influência das variáveis capital investido $(x_1)$ e gasto em publicidade $(x_2)$ no lucro anual (y) de empresas, foram observadas essas variáveis em doze empresas em um mesmo ano. Os seguintes resultados foram registrados, na unidade de 100.000 reais.

|  y | x1 | x2 |
|:--:|:-----:|:-----:|
| 12 |   31  |   4   |
| 13 |   16  |   5   |
|  3 |   29  |   3   |
|  3 |   19  |   0   |
| 11 |   27  |   2   |
| 19 |   21  |   6   |
|  1 |   24  |   2   |
| 14 |   11  |   3   |
| 15 |   26  |   6   |
| 17 |   18  |   6   |
|  2 |   12  |   1   |
| 15 |   3   |   5   |

+ a) Ajuste a estes dados um modelo do tipo: $y=\beta_0+\beta_1x_1+\beta_2x_2+ \epsilon$.
+ b) Obtenha a tabela de ANOVA;
+ c) Encontre $R^2$ e interprete o seu valor;
+ d) Os dados fornecem evidências suficientes para indicar que o modelo contribui significativamente para a predição de y, ao nível de 5\% de significância?
+ e) Teste as hipóteses $H_0:\beta_2 = 0$ versus $H_1: \beta_2 \neq 0$, ao nível de 5\% de significância?

$\color{red}{\text{a)}}$ - **Ajuste a estes dados um modelo do tipo: $y=\beta_0+\beta_1x_1+\beta_2x_2+\epsilon$.**

In [3]:
#Entrada de dados
q8.1=read.csv2("investimento.csv",header=T,dec=".")

#Ajuste ao modelo
ajuste8.1=lm(y~x1+x2,data=q8.1)
summary(ajuste8.1)


Call:
lm(formula = y ~ x1 + x2, data = q8.1)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.8043 -1.8136 -0.4532  1.9866  5.5675 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   3.4022     3.4221   0.994 0.346105    
x1           -0.1180     0.1313  -0.899 0.392241    
x2            2.6077     0.5277   4.941 0.000801 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.613 on 9 degrees of freedom
Multiple R-squared:  0.7395,	Adjusted R-squared:  0.6816 
F-statistic: 12.77 on 2 and 9 DF,  p-value: 0.002351


O modelo ajustado é $\hat{y}=3,4022-0,1180X_1+2,6077X_2+\epsilon$.

$\color{red}{\text{b)}}$ - **Obtenha a tabela de ANOVA**

In [4]:
anova(ajuste8.1)

Df Sum Sq    Mean Sq   F value   Pr(>F)      
x1        1   14.74852  14.74852  1.129933 0.3154783274
x2        1  318.69508 318.69508 24.416283 0.0008008264
Residuals 9  117.47307  13.05256        NA           NA

O parâmetro $\beta_2$ é considerado significativo, assim como a regressão.

$\color{red}{\text{c)}}$ - **Encontre $R^2$ e interprete o seu valor**

In [5]:
# Primeiro Método
r2=summary(ajuste8.1)
r2_adj=r2$adj.r.squared
cat("O coeficiente de determinação ajustado é dado por",round(r2_adj*100,2),"%","\n\n")



O coeficiente de determinação ajustado é dado por 68.16 % 



In [6]:
#Segundo Método : R2=SQ_regressao/ SQ_tot
#SQ_regressao = t(beta) . t(X) . y - n . media(y)^2
#SQ_erro= t(y) . y - t(beta) . t(X) . y
#SQ_total = t(y) . y - n . media(y)^2

#Base de dados
x1=cbind(q8.1$x1)        #Extraíndo a coluna x1
x2=cbind(q8.1$x2)        #Extraíndo a coluna x2
y=cbind(q8.1$y)          #Extraíndo a coluna y

#Calculado SQ_total
SQ_total = t(y) %*% y - length(y) %*% mean(y)^2

#Criando a matrix X
x=matrix(c(rep(1,12),x1,x2),ncol=3,byrow=F)

#Determinando Beta
beta=solve(t(x)%*%x)%*%t(x)%*%y

#Determinando SQ_regressao
SQ_regressao= t(beta) %*% t(x) %*% y - length(y) %*% mean(y)^2

#Determinando o R2
R2_cal=SQ_regressao/SQ_total
cat("O coeficiente de determinação é",R2_cal,"\n\n")

#Soma dos quadrados do Erro
SQ_erro=t(y) %*% y - t(beta) %*% t(x) %*% y

#Número de parâmetros
p=2

#Coeficiente de determinação ajustado
R2_ajus=1-((SQ_erro)/(length(y)-p-1))/(SQ_total/(length(y)-1))
cat("O coeficiente de determinação ajustado é",R2_ajus)

O coeficiente de determinação é 0.7394794 

O coeficiente de determinação ajustado é 0.681586

$\color{red}{\text{d)}}$ - **Os dados fornecem evidências suficientes para indicar que o modelo contribui significativamente para a predição de y, ao nível de 5\% de significância?**

O modelo é considerado significativo ao nível de 5\% de significância.

$\color{red}{\text{e)}}$ - **Teste as hipóteses $H_0:\beta_2=0$ versus $H_1:\beta_2 \neq 0$, ao nível de 5\% de significância?**

O coeficiente $\beta_2$ é significativo para o modelo ao nível de 5\% de significância, visto que o p-valor para análise isolada de seu coeficiente é de 0.000801, rejeitando-se desta forma a hipótese $H_0$.

$\color{red}{\text{Exercício 2}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** Uma indústria química está interessada em maximizar a sua produção de um certo processo químico. Para isso, obteve os seguintes resultados da temperatura codificada, x, e de peso do produto, y, em kg:

| x |  -5 | -4 |  -3 |  -2 |  -1  |  0 |   1  |   2  |  3  | 4 |  5  |
|:-:|:---:|:--:|:---:|:---:|:----:|:--:|:----:|:----:|:---:|:-:|:---:|
| y | 6.4 |  7 | 7.6 | 8.3 | 10.4 | 12 | 11.3 | 10.6 | 9.9 | 9 | 8.3 |

Neste tipo de reação química, a quantidade da substância produzida pode aumentar até um certo valor crítico, no qual a quantidade da substância produzida começa a diminuir devido à sua decomposição pelo aumento da temperatura.

+ a) Ajuste o modelo $y=\beta_0+\beta_1x+\beta_2x^2+ \epsilon$;
+ b) Encontre $R^2$ e interprete o seu valor;
+ c) Há evidências suficientes para indicar que o modelo contribui significamente na predição de y, ao nível de 5\%? Justifique a sua resposta;
+ d) Teste as hipóteses $H_0: \beta_2 = 0$ versus $H_1:\beta_2 \neq 0$, com $\alpha=0,05$;
+ e) Obtenha os valores de y preditos para $x=-1$ e $x=3$.

$\color{red}{\text{a)}}$ - **Ajuste o modelo $y=\beta_0+\beta_1x+\beta_2x^2+\epsilon$**

In [7]:
###### Entrando com os dados
x=seq(-5,5,by=1)
x2=x^2
y=c(6.4, 7, 7.6, 8.3, 10.4, 12, 11.3, 10.6, 9.9, 9, 8.3)
base=data.frame(y,x,x2)

#Ajustando o modelo
ajuste8.2=lm(y~x+x2,data=base)
summary(ajuste8.2)



Call:
lm(formula = y ~ x + x2, data = base)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.22699 -0.28709 -0.01427  0.32587  1.32471 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 10.67529    0.37045  28.817 2.28e-09 ***
x            0.27182    0.07755   3.505 0.008018 ** 
x2          -0.15117    0.02777  -5.444 0.000613 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.8133 on 8 degrees of freedom
Multiple R-squared:  0.8398,	Adjusted R-squared:  0.7997 
F-statistic: 20.96 on 2 and 8 DF,  p-value: 0.0006593


O modelo estimado é dado por $\hat{y}=10,67+0,2718x_1-0,1511x_2+\epsilon$ com um coeficiente de determinação ajustado de 79,97\%.

$\color{red}{\text{b)}}$ - **Encontre $R^2$ e interprete o seu valor**

In [8]:
#Método 1 - R2 - Ajustado
nome=summary(ajuste8.2)
Coef_adj=nome$adj.r.squared

cat("\n\nO coeficiente de determinação ajustado para o modelo proposto é",round(Coef_adj*100,2),"%","\n\n")



O coeficiente de determinação ajustado para o modelo proposto é 79.97 % 



$\color{red}{\text{c)}}$ - **Há evidências suficientes para indicar que o modelo contribui significativamente na predição de y, ao nível de 5\%? Justifique a sua respsota.$**

O modelo apresenta todos os parâmetros significativos para compor o modelo, indicando que o modelo é significativo.

$\color{red}{\text{d)}}$ - **Teste as hipóteses $H_0: \beta_2 = 0$ versus $H_1: \beta_2 \neq 0$, com $\alpha=0,05$.**

O parâmetro $\beta_2$ é extremamente significativo para o modelo, visto que o p-valor associado ao teste do parâmetro é de 0.000613, rejeitando-se a hipótese nula e aceitando-se a hipotese alternativa indicando a significância do parâmetro para o modelo.

$\color{red}{\text{e)}}$ - **Obtenha os valores de y preditos para x=-1 e x=3**

In [9]:
novos=data.frame(x=c(-1,3),x2=c(1,9))
predict(ajuste8.2,novos,interval="prediction",level=0.95)

fit      lwr      upr     
1 10.25231 8.202572 12.30204
2 10.13026 8.098198 12.16231

Os valores previstos para $x_1=-1$ e $x_2=1$ substituídos no modelo estimado $\hat{y}=10,67+0,2718x_1-0,1511x_2+\epsilon$ foi obtido da seguinte forma:

$$\hat{y}=10,67+0,2718x_1-0,1511x_2+\epsilon$$
$$\hat{y}=10,67+0,2718\cdot (-1) -0,1511 \cdot (1) = 10,2531$$

$\color{red}{\text{Exercício 3}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** Um experimento foi realizado para se estudar a relação entre o grau de corrosão de um certo metal, y, e o tempo de exposição (em semanas), x, desse metal à ação da acidez do solo. Foram obtidos os seguintes resultados:

| x |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |  10  |
|:-:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|
| y | 0.08 | 0.18 | 0,32 | 0.53 | 0.88 | 1.30 | 1.95 | 2.80 | 3.90 | 4.60 |

+ a) Ajuste a esses dados o modelo $y=\beta_0+\beta_1x+\beta_2x^2+\epsilon$;
+ b) Obtenha a tabela de ANOVA;
+ c) O modelo ajustado é significante ao nível de 5\%?
+ d) O coeficiente de regressão linear é significante ao nível de 5\%?
+ e) Que porcentagem da variação total é explicada pela regressão quadrática de y em x?

$\color{red}{\text{a)}}$ - **Ajuste a esses dados o modelo $y=\beta_0+\beta_1x+\beta_2x^2+\epsilon$**

In [10]:
#Entrando com os dados
x=seq(1:10)
x2=x^2
y=c(0.08, 0.18, 0.32, 0.53, 0.88, 1.30, 1.95, 2.80, 3.90, 4.60)
q8.3=data.frame(y,x,x2)


#Ajuste do modelo
ajuste8.3=lm(y~x+x2,data=q8.3)
summary(ajuste8.3)


Call:
lm(formula = y ~ x + x2, data = q8.3)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.110364 -0.075136 -0.000667  0.040773  0.211455 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.274000   0.127775   2.144  0.06918 .  
x           -0.198788   0.053364  -3.725  0.00741 ** 
x2           0.064242   0.004728  13.588 2.75e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.1086 on 7 degrees of freedom
Multiple R-squared:  0.9965,	Adjusted R-squared:  0.9955 
F-statistic: 993.8 on 2 and 7 DF,  p-value: 2.56e-09


O modelo estimado é dado por $\hat{y}=0,274-0,198788x_1+0,064242x_2+\epsilon$.

$\color{red}{\text{b)}}$ - **Obtenha a tabela de ANOVA**

In [11]:
anova(ajuste8.3)

Df Sum Sq      Mean Sq     F value   Pr(>F)      
x         1  21.28012121 21.28012121 1803.0561 1.048352e-09
x2        1   2.17910303  2.17910303  184.6345 2.750318e-06
Residuals 7   0.08261576  0.01180225        NA           NA

O modelo estimado é significativo para a realização de predições. Os parâmetros estimados para o modelo são significativos.

$\color{red}{\text{c)}}$ - **O modelo ajustado é significante ao nível de 5\%?**

O modelo estimado é significativo ao nível de 5\% de significância.


$\color{red}{\text{d)}}$ - **O coeficiente de regressão linear é significante ao nível de 5\%?**

Em um modelo de regressão linear o parâmetro $\beta_0$ é chamado de intercepto ou coeficiente linear e o parâmetro $\beta_1$ é chamado coeficiente de regressão ou coeficiente angular. Os coeficientes estimados para o modelo são significativos para a realização de predições.

$\color{red}{\text{e)}}$ - **Que porcentagem da variação total é explicada pela regressão quadrática de y em x?** 

O Coeficiente de determinação ajustado para o modelo estimado é de $R^2_{adj}=99,55\%$, indicando que o modelo estimado explica em grande porcentagem a variação de  y em x.

$\color{red}{\text{Exercício 4}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** Considere as seguintes observações:

| Operador |  y | x1 |  x2 |
|:--------:|:--:|:--:|:---:|
|     1    | 12 | 18 |  2  |
|     2    |  3 | 16 |  3  |
|     3    | 11 | 25 |  2  |
|     4    |  1 | 12 |  3  |
|     5    | 13 | 20 |  3  |
|     6    | 20 | 35 |  2  |
|     7    |  2 | 17 | 1.5 |
|     8    | 25 | 25 |  5  |
|     9    | 26 | 39 |  1  |
|    10    | 15 | 20 | 2.5 |
|    11    |  1 | 18 |  2  |
|    12    | 15 | 29 | 3.5 |
|    13    | 11 | 20 |  5  |
|    14    |  5 | 16 |  1  |
|    15    |  7 | 29 | 1.5 |

Onde:

y: número de itens produzidos com defeito, em certo dia;

x1: produção média por hora;

x2: tempo, em semanas, decorrido desde o último reparo na máquina.

+ a) Ajuste um modelo para o número de itens produzidos com defeito em função da produção média por hora e do tempo decorrido desde o último reparo na máquina;
+ b) Encontre e interprete o valor de $R^2$;
+ c) Pode-se afirmar que há evidências suficientes de que o modelo encontrado em (a) contribua na predição de y? (use $\alpha=1\%$).

$\color{red}{\text{a)}}$ - **Ajuste um modelo para o número de itens produzidos com defeito em função da produção média por hora e do tempo decorrido desde o último reparo na máquina**

In [12]:
#Entrada de dados
operador=read.csv2("operador.csv",header=T,dec=".")

#Ajuste do modelo
ajusteq4=lm(y~x1+x2,data=operador)
summary(ajusteq4)


Call:
lm(formula = y ~ x1 + x2, data = operador)

Residuals:
   Min     1Q Median     3Q    Max 
-7.417 -3.445 -1.028  3.566  6.247 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -14.7406     5.1814  -2.845 0.014761 *  
x1            0.8860     0.1708   5.186 0.000227 ***
x2            2.3094     1.0427   2.215 0.046880 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 4.792 on 12 degrees of freedom
Multiple R-squared:  0.7055,	Adjusted R-squared:  0.6564 
F-statistic: 14.37 on 2 and 12 DF,  p-value: 0.0006522


O modelo estimado é dado por $\hat{y}=-14,7406+0,8860x_1+2,3094x_2$.

$\color{red}{\text{b)}}$ - **Encontre e interprete o valor de $R^2$**.

In [13]:
r2q5=summary(ajusteq4)
cat("\n\nO valor do coeficiente de determinação é",r2q5$r.squared,"e o coeficiente de determinação ajustado é de",r2q5$adj.r.squared,"\n\n")



O valor do coeficiente de determinação é 0.7055179 e o coeficiente de determinação ajustado é de 0.6564376 



$\color{red}{\text{c)}}$ - **Pode-se afirmar que há evidências suficientes de que o modelo encontrado em (a) contribua na predição de y? (use $\alpha=1\%$)**

O modelo estimado possuem parâmetros significativos ao nível de 1\% de significância, indicando que contribuem para que o modelo seja considerado significativo para a realização de predições de y.

$\color{red}{\text{Exercício 5}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** Um experimento foi realizado para se verificar o efeito da tempertura $(x_1)$ e da concentração $(x_2)$ na produção (y) de um certo processo químico. Os dados obtidos foram os seguintes:

|  y  |  x1 | x2 |  y  |  x1 | x2 |
|:---:|:---:|:--:|:---:|:---:|:--:|
| 189 |  80 | 10 | 220 |  80 | 20 |
| 203 | 100 | 10 | 228 | 100 | 20 |
| 222 | 120 | 10 | 252 | 120 | 20 |
| 234 | 140 | 10 | 263 | 140 | 20 |
| 261 | 160 | 10 | 291 | 160 | 20 |
| 204 |  80 | 15 | 226 |  80 | 25 |
| 212 | 100 | 15 | 232 | 100 | 25 |
| 223 | 120 | 15 | 259 | 120 | 25 |
| 246 | 140 | 15 | 268 | 140 | 25 |
| 273 | 160 | 15 | 294 | 160 | 25 |

+ a) Ajuste um modelo que explique a quantidade do produto em função da temperatura e da concentração da substância no processo químico;
+ b) Encontre um intervalo de confiança para o coeficiente da variável concentração no modelo ajustado;
+ c) Calcule e interprete o valor do coeficiente de determinação para esse modelo;
+ d) Estime a variância da variável dependente no modelo ajustado.

$\color{red}{\text{a)}}$ - **Ajuste um modelo que explique a quantidade do produto em função da temperatura e da concentração da substância no processo químico**

In [14]:
#Entrada de dados
q5=read.csv2("temp.csv",header=T,dec=".")

ajusteq5=lm(y~x1+x2,data=q5)
summary(ajusteq5)


Call:
lm(formula = y ~ x1 + x2, data = q5)

Residuals:
   Min     1Q Median     3Q    Max 
-10.94  -4.71   0.50   4.29  10.14 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 93.18000    7.25415   12.85 3.53e-10 ***
x1           0.87000    0.04779   18.20 1.38e-12 ***
x2           2.42400    0.24180   10.03 1.49e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 6.045 on 17 degrees of freedom
Multiple R-squared:  0.9621,	Adjusted R-squared:  0.9577 
F-statistic: 215.9 on 2 and 17 DF,  p-value: 8.234e-13


O modelo ajustado é dado por $\hat{y}=93,18+0,87x_1+2,424x_2$.

$\color{red}{\text{b)}}$ - **Encontre um intervalo de confiança para o coeficiente da variável concentração no modelo ajustado**

In [15]:
confint(ajusteq5,level=0.95)

2.5 %     97.5 %    
(Intercept) 77.875083 108.484917
x1           0.769170   0.970830
x2           1.913836   2.934164

O intervalo de confiança para o parâmetro $\beta_2$ é dado por (1,913836 ; 2,934164) ao nível de 95\% de confiança.

$\color{red}{\text{c)}}$ - **Calcule e interprete o valor do coeficiente de determinação para esse modelo**

In [16]:
#Determinação do coeficiente de determinação - Método 1
met1=summary(ajusteq5)
cat("O coeficiente de determinação é dado por",met1$r.squared,"e o coeficiente de determinação ajustado é dado por",met1$adj.r.squared)


O coeficiente de determinação é dado por 0.9621287 e o coeficiente de determinação ajustado é dado por 0.9576733

In [17]:
#Determinação do coeficiente de determinação - Método 2


#Segundo Método : R2=SQ_regressao/ SQ_tot
#SQ_regressao = t(beta) . t(X) . y - n . media(y)^2
#SQ_erro= t(y) . y - t(beta) . t(X) . y
#SQ_total = t(y) . y - n . media(y)^2

#Base de dados
x1=cbind(q5$x1)        #Extraíndo a coluna x1
x2=cbind(q5$x2)        #Extraíndo a coluna x2
y=cbind(q5$y)          #Extraíndo a coluna y

#Calculado SQ_total
SQ_total = t(y) %*% y - length(y) %*% mean(y)^2

#Criando a matrix X
x=matrix(c(rep(1,20),x1,x2),ncol=3,byrow=F)

#Determinando Beta
beta=solve(t(x)%*%x)%*%t(x)%*%y

#Determinando SQ_regressao
SQ_regressao= t(beta) %*% t(x) %*% y - length(y) %*% mean(y)^2

#Determinando o R2
R2_cal=SQ_regressao/SQ_total
cat("O coeficiente de determinação é",R2_cal,"\n\n")

#Soma dos quadrados do Erro
SQ_erro=t(y) %*% y - t(beta) %*% t(x) %*% y

#Número de parâmetros
p=2

#Coeficiente de determinação ajustado
R2_ajus=1-((SQ_erro)/(length(y)-p-1))/(SQ_total/(length(y)-1))
cat("O coeficiente de determinação ajustado é",R2_ajus)


O coeficiente de determinação é 0.9621287 

O coeficiente de determinação ajustado é 0.9576733

$\color{red}{\text{d)}}$ - **Estime a variância da variável dependente no modelo ajustado**

In [18]:
#Cálculo da variância - Primeiro Método
met1=summary(ajusteq5)
sigma=met1$sigma
cat("A variância da variável dependente no modelo é dado por",sigma^2)


A variância da variável dependente no modelo é dado por 36.54353

In [19]:
#Cálculo da variância - Método 2
#sigma2=(t(y-X.beta)) . (y-X.beta))/(n-p-1)

sigma2=(t(y-x%*%beta) %*% (y-x%*%beta))/(length(y)-p-1)
cat("Pelo fórmula, a variância é dada por",sigma2)

Pelo fórmula, a variância é dada por 36.54353

$\color{red}{\text{Exercício 6}}$ - **Retirado do livro: Análise de Modelos de Regressão Linear com Aplicações, de Charnet, Freire, Charnet e Bonvino** Na tabela a seguir estão apresentados alguns resultados do Vestibular 98 da Unicamp, por curso. As variáveis apresentadas são:

+ Relação candidato/vaga (C/V) para a primeira fase do Vestibular;
+ Notas médias das provas de Química (NQ), Matemática (NM) e História (NH), na segunda fase do Vestibular;
+ Período (Per) em que é oferecido o curso: diurno (D) ou noturno (V);
+ Disciplina prioritária para o curso, Química (QP), Matemática (MP) e História (HP): sim (S) ou não (N).

|         Cursos        | Per. |  C/V  |   NQ  | QP |   NM  | MP |   NH  | HP |
|:---------------------:|:----:|:-----:|:-----:|:--:|:-----:|:--:|:-----:|:--:|
|     Artes cênicas     |   D  | 15.08 | 22.06 |  N | 15.92 |  N | 25.11 |  N |
|     C. Computação     |   V  | 29.60 | 29.33 |  N | 24.41 |  N | 23.94 |  N |
|     C. biológicas     |   D  | 37.83 | 31.18 |  N | 20.21 |  N | 26.07 |  N |
|     C. biológicas     |   V  |  9.45 | 24.61 |  N | 15.54 |  N | 21.88 |  N |
|     C. econômicas     |   D  | 16.94 | 26.74 |  N | 22.22 |  S | 28.96 |  S |
|     C. econômicas     |   V  | 12.43 | 25.43 |  N | 21.05 |  S | 28.53 |  S |
|    Ciências sociais   |   D  |  6.92 | 22.24 |  N | 15.73 |  N | 29.47 |  S |
|    Ciências sociais   |   V  |  6.22 | 13.42 |  N | 10.91 |  N | 24.33 |  S |
|         Dança         |   D  |  6.64 | 13.98 |  N | 10.78 |  N | 18.05 |  N |
|     Ed. artística     |   D  |  8.60 | 16.77 |  N | 13.54 |  N | 22.01 |  N |
|    Educação física    |   D  | 13.26 | 23.10 |  N | 16.02 |  N | 21.20 |  N |
|    Educação física    |   V  |  7.66 | 15.10 |  N | 11.92 |  N | 17.59 |  N |
|       Enfermagem      |   D  | 20.43 | 27.04 |  N | 17.19 |  N | 24.16 |  N |
|      E. agrícola      |   D  |  5.93 | 20.66 |  N | 15.39 |  S | 19.95 |  N |
|      E. alimentos     |   D  | 17.70 | 31.81 |  N | 23.25 |  S | 24.47 |  N |
|      E. alimentos     |   V  | 16.73 | 30.78 |  N | 21.75 |  S | 23.09 |  N |
|        E. civil       |   D  | 13.19 | 30.34 |  N | 25.89 |  S | 24.97 |  N |
|    E. c. automação    |   V  | 26.08 | 30.09 |  N | 23.81 |  S | 23.66 |  N |
|     E. computação     |   D  | 20.02 | 34.53 |  N | 27.41 |  S | 25.42 |  N |
|      E. elétrica      |   D  | 15.44 | 34.89 |  N | 27.84 |  S | 25.34 |  N |
|      E. elétrica      |   V  | 13.80 | 26.65 |  N | 23.94 |  S | 22.12 |  N |
|      E. mecânica      |   D  | 17.10 | 33.57 |  N | 26.55 |  S | 25.25 |  N |
|       E. química      |   D  | 10.62 | 36.20 |  S | 25.72 |  S | 24.70 |  N |
|       E. química      |   V  |  7.20 | 35.21 |  S | 19.79 |  S | 22.18 |  N |
|      Estatística      |   D  |  4.31 | 13.76 |  N | 12.19 |  S | 14.88 |  N |
|       Filosofia       |   D  |  4.40 | 13.11 |  N | 12.16 |  S | 21.53 |  S |
|         Física        |   V  |  4.40 | 17.95 |  N | 13.97 |  S | 17.50 |  N |
|    F/M/M. aplicada    |   D  |  3.39 | 26.57 |  N | 22.90 |  S | 21.41 |  N |
|       Geografia       |   V  |  3.07 | 11.21 |  N |  9.15 |  N | 20.48 |  N |
|   Geologia/Geografia  |   D  |  6.07 | 16.92 |  N | 11.84 |  N | 22.29 |  N |
|        História       |   D  | 11.83 | 21.16 |  N | 15.17 |  N | 30.28 |  S |
|  Letras e linguística |   D  |  7.56 | 17.99 |  N | 12.80 |  N | 24.28 |  S |
|   Mat. licenciatura   |   V  |  2.89 | 12.39 |  N | 12.38 |  S | 14.46 |  N |
|        Medicina       |   D  | 84.34 | 40.41 |  S | 28.58 |  N | 31.63 |  N |
|   Mús. - composição   |   D  |  2.30 | 14.67 |  N | 10.83 |  N | 18.50 |  N |
|   Mús. - instrumento  |   D  |  4.00 |  9.37 |  N |  7.86 |  N | 13.05 |  N |
|     Mús. - popular    |   D  |  9.00 | 22.20 |  N | 16.85 |  N | 24.22 |  N |
|    Mús. - regência    |   D  |  2.30 | 12.00 |  N |  9.39 |  N | 17.56 |  N |
|      Odontologia      |   D  | 30.54 | 31.18 |  S | 20.53 |  N | 24.72 |  N |
|       Pedagogia       |   D  |  5.27 | 11.49 |  N |  9.01 |  N | 18.87 |  S |
|       Pedagogia       |   V  | 10.10 | 10.61 |  N |  9.43 |  N | 18.85 |  S |
|        Química        |   D  |  5.36 | 32.92 |  S | 19.06 |  S | 22.37 |  N |
| Quím. modal. tecnol.  |   V  |  8.03 | 33.39 |  S | 15.52 |  N | 18.74 |  N |
|     Tec. c. civil     |   V  |  1.64 | 12.85 |  N | 10.55 |  S | 14.82 |  N |
|     Tec. p. dados     |   V  |  6.38 | 13.97 |  N | 12.09 |  S | 17.35 |  N |
|     Tec. sanitária    |   V  |  3.46 | 11.74 |  N |  8.15 |  S | 12.88 |  N |


Para cada uma das disciplinas (Química, Matemática e História) encontre o melhor modelo para relacionar a nota obtida na prova do Vestibular e as demais variáveis.

In [20]:
#Entrada de dados
cursos=read.csv2("cursos.csv",header=T,dec=".")
head(cursos)


Cursos        Per CV    NQ    QP NM    MP NH    HP
1 Artes cênicas D   15.08 22.06 N  15.92 N  25.11 N 
2 C. Computação V   29.60 29.33 N  24.41 N  23.94 N 
3 C. biológicas D   37.83 31.18 N  20.21 N  26.07 N 
4 C. biológicas V    9.45 24.61 N  15.54 N  21.88 N 
5 C. econômicas D   16.94 26.74 N  22.22 S  28.96 S 
6 C. econômicas V   12.43 25.43 N  21.05 S  28.53 S

In [21]:
#filtros turnos
require(dplyr)                                      
per_diurno=cursos %>% filter(Per=="D")
per_not=cursos %>% filter(Per=="V")

#filtros QP prioritária
qpS=cursos %>% filter(QP=="S")
qpN=cursos %>% filter(QP=="N")


Loading required package: dplyr

Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union



In [22]:
ajuste_quimica=lm(NQ~CV,data=per_diurno)
summary(ajuste_quimica)


Call:
lm(formula = NQ ~ CV, data = per_diurno)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.9449  -5.1248  -0.3953   6.4646  13.6115 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 18.94104    1.73443  10.921 2.08e-11 ***
CV           0.34345    0.08267   4.155 0.000294 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 6.951 on 27 degrees of freedom
Multiple R-squared:   0.39,	Adjusted R-squared:  0.3674 
F-statistic: 17.26 on 1 and 27 DF,  p-value: 0.0002937


O modelo obtido é dado por $\hat{y}=18,94+0,34345CV$ para os alunos de química diurno com um coeficiente de determinação de 39\%.